### PR TITLE
108_Convert_Sorted_Array_to_Binary_Search_Tree.py

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code

### CODE REVIEW

File: 108_Convert_Sorted_Array_to_Binary_Search_Tree.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Time: O(logn)
+# Space: O(n)
+from typing import List
+
+
+class TreeNode:
+    
+    def __init__(self, val=0, left=None, right=None):
```
Comment (ryujumoto) [inline 108_Convert_Sorted_Array_to_Binary_Search_Tree.py]: Add type annotation.

File: 108_Convert_Sorted_Array_to_Binary_Search_Tree.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Solution:
+    
+    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
+        def createNode(start, end, nums):
```
Comment (ryujumoto) [inline 108_Convert_Sorted_Array_to_Binary_Search_Tree.py]: Add type annotation.
Also, functions should be named according to the snake case naming convention.

File: 108_Convert_Sorted_Array_to_Binary_Search_Tree.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Time: O(logn)
+# Space: O(n)
```
Comment (ryujumoto) [inline 108_Convert_Sorted_Array_to_Binary_Search_Tree.py]: The time complexity and space complexity are wrong.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
@Ryuju0911 
I added the type annotation and name the func ad 'create_Node' according to your advice.
Please check my code.

File: 108_Convert_Sorted_Array_to_Binary_Search_Tree.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    def __init__(self, val=0, left=None, right=None) -> None:
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Solution:
+    
+    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
+        def create＿Node(start, end, nums) -> Optional[TreeNode]:
```
Comment (ryujumoto) [inline 108_Convert_Sorted_Array_to_Binary_Search_Tree.py]: `def create_node(start: int, end: int, nums: List[int]) -> Optional[TreeNode]:`
You have to write the type annotations for the arguments, too.

File: 108_Convert_Sorted_Array_to_Binary_Search_Tree.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Time: O(nlogn)
+# Space: O(1)
+from typing import List, Optional
+
+
+class TreeNode:
+    
+    def __init__(self, val=0, left=None, right=None) -> None:
```
Comment (ryujumoto) [inline 108_Convert_Sorted_Array_to_Binary_Search_Tree.py]: def __init__(self, val: int = 0, left: Optional[TreeNode] =  None, right: Optional[TreeNode] = None) -> None:

**Review by ryujumoto — COMMENTED** [review]
Check my review.

**Comment by Rwatana** [comment]
I think I revised them and learned how to make annotations. Please check my code.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**